libs/clients/electronic-registration-statistics/src/lib/electronicRegistrations.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Inject, Injectable } from '@nestjs/common'
+import { Configuration, StatisticsApi } from '../../gen/fetch'
+import { ElectronicRegistrationsClientConfig } from './electronicRegistrations.config'
+import type { ConfigType } from '@island.is/nest/config'
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+
+@Injectable()
+export class ElectronicRegistrationsClientService {
+  constructor(
+    @Inject(ElectronicRegistrationsClientConfig.KEY)
+    private clientConfig: ConfigType<
+      typeof ElectronicRegistrationsClientConfig
+    >,
+  ) {}
+
+  private async createApi() {
+    return new StatisticsApi(
+      new Configuration({
+        fetchApi: createEnhancedFetch({
+          name: 'clients-electronic-registration-statistics',
+          ...this.clientConfig.fetch,
+        }),
+        basePath: this.clientConfig.url,
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+      }),
+    )
```

Comment:
This should be a singleton, provided in the NestJS module and injected as a dependency here. Like in most other client libraries.

Reply:
I changed the code :)

Reply:
Figures. The syslumenn client does this because they need custom login logic to get an access token.